thewalrus/_permanent.py

```diff
@@ -68,34 +63,106 @@ def perm(A, quad=True, fsum=False, method="bbfg"):
 
     isRyser = bool(method != "bbfg")
 
-    if A.dtype == np.complex:
-        if np.any(np.iscomplex(A)):
-            return perm_complex(A, quad=quad) if isRyser else perm_BBFG_complex(A)
-        return (
-            perm_real(np.float64(A.real), quad=quad, fsum=fsum)
-            if isRyser
-            else perm_BBFG_real(np.float64(A.real))
-        )
-    return perm_real(A, quad=quad, fsum=fsum) if isRyser else perm_BBFG_real(A)
+    return perm_ryser(A) if isRyser else perm_bbfg(A)
+
+
+@jit(nopython=True)
+def perm_ryser(M):
```

Comment:
I've just noticed that `perm_ryser` doesn't show up in the documentation 🙂 https://hafnian--300.org.readthedocs.build/en/300/code/thewalrus.html

Is this intentional? If not, it probably needs to be listed above in the module docstring somewhere

Reply:
```suggestion
def perm_ryser(M):  # pragma: no cover
```